src/js/route.js

```diff
@@ -6,17 +6,28 @@ class Router extends String {
 
         this.name           = name;
         this.absolute       = absolute;
+        this.template       = this.name ? new UrlBuilder(name, absolute).construct() : '',
         this.urlParams      = this.normalizeParams(params);
         this.queryParams    = this.normalizeParams(params);
     }
 
     normalizeParams(params) {
-        if (typeof params === 'undefined')
+	if (typeof params === 'undefined')
             return {};
 
+        // If you passed in a string or integer, wrap it in an array
         params = typeof params !== 'object' ? [params] : params;
-        this.numericParamIndices = Array.isArray(params);
+        
+        // If the tags object contains an ID and there isn't an ID param in the
+        // url template, they probably passed in a single model object and we should
+        // wrap this in an array. This could be slightly dangerous and I want to find
+        // a better solution for this rare case.
+        
+        if (params.hasOwnProperty('id') && this.template.indexOf('{id}') == -1) {
+            params = [params.id];
+        }
 
+        this.numericParamIndices = Array.isArray(params);
```

Comment:
this is not working if there are route modal bindings and query paramaters (product/{product}/edit?someParam=123)